### PR TITLE
feat: add Header component and theme

### PR DESCRIPTION
**Pull request by loc-nguyenthien**
## Summary
- Adding Header component
- Adding custom theme for Header

## Check List
- [ ] Your code builds clean without any errors or warnings.
- [ ] Check coding convention.
- [ ] Updated Storybook components and stories to reflect any changes made to the UI.
- [ ] Test covers all edge cases and coverage is greater than or equal to 80%.

## Proof of Completeness
Simulator Screen Shot - iPhone 11 - 2023-11-09 at 09 58 48

### CODE REVIEW

File: src/components/Header/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      backgroundColor={backgroundColor ?? HeaderTheme.backgroundColor}
+      height={(height ?? (HeaderTheme.height as number)) + (paddingTop ?? (HeaderTheme.paddingTop as number))}
+      paddingTop={paddingTop ?? HeaderTheme.paddingTop}
+      borderBottomWidth={borderBottomWidth ?? HeaderTheme.borderBottomWidth}
+      borderBottomColor={borderBottomColor ?? HeaderTheme.borderBottomColor}
+      {...StyleSheet.flatten(containerStyle)}>
+      <SidesContainer {...StyleSheet.flatten(leftStyle)}>
+        {leftComponent
+          ? leftComponent
+          : !!hasBackButton && (
```
Comment (baonguyenhsts) [inline src/components/Header/Header.tsx:117]: remove !!
```suggestion
          : hasBackButton && (
```

File: src/components/Header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  centerStyle,
+  rightComponent,
+  rightStyle,
+}) => {
+  const HeaderTheme = useTheme().components.Header
+  return (
+    <Container
+      backgroundColor={backgroundColor ?? HeaderTheme.backgroundColor}
+      height={(height ?? (HeaderTheme.height as number)) + (paddingTop ?? (HeaderTheme.paddingTop as number))}
+      paddingTop={paddingTop ?? HeaderTheme.paddingTop}
```
Comment (baonguyenhsts) [inline src/components/Header/Header.tsx:110]: The `HeaderTheme.paddingTop` is metrics.xxl, you don't use the insets from `react-native-safe-area-context`

**Review by baonguyenhsts — CHANGES_REQUESTED** [review]

File: src/components/Header/Header.tsx (hunk 2 above)
Comment (loc-nguyenthien) [reply src/components/Header/Header.tsx:110]: That is because I don't want to add `react-native-safe-area-context` to our package. I mentioned it in the comment of the props above
Screenshot 2023-11-09 at 10 26 21

Comment (loc-nguyenthien) [reply src/components/Header/Header.tsx:110]: My purpose was something like this
Screenshot 2023-11-09 at 11 31 36

File: src/components/Header/Header.tsx (hunk 1 above)
Comment (loc-nguyenthien) [reply src/components/Header/Header.tsx]: Updated

File: src/components/Header/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+   */
+  centerStyle?: StyleProp<ViewStyle>
+  /**
+   * The right component of the Header
+   */
+  rightComponent?: ReactNode
+  /**
+   * The style of right component of the Header
+   */
+  rightStyle?: StyleProp<ViewStyle>
```
Comment (ledutu2) [inline src/components/Header/Header.tsx:82]: Do we need the comment above every props? I think the variable name is quite clear to understand.

**Review by ngochuyduong — APPROVED** [review]

File: src/components/Header/Header.tsx (hunk 3 above)
Comment (loc-nguyenthien) [reply src/components/Header/Header.tsx:82]: I commented to keep the convention with other previous components, which we also wrote comments.

**Review by chungnguyenhoang — APPROVED** [review]

**Review by baonguyenhsts — APPROVED** [review]

**Review by ledutu2 — APPROVED** [review]
